packages/block-library/src/post-title/edit.js

```diff
@@ -20,15 +21,14 @@ import { useEntityProp } from '@wordpress/core-data';
 /**
  * Internal dependencies
  */
-import HeadingLevelDropdown from '../heading/heading-level-dropdown';
 import { useCanEditEntity } from '../utils/hooks';
 
 export default function PostTitleEdit( {
 	attributes: { level, textAlign, isLink, rel, linkTarget },
 	setAttributes,
 	context: { postType, postId, queryId },
 } ) {
-	const TagName = 0 === level ? 'p' : 'h' + level;
```

Comment:
I noticed that this code seems to have been adapted for allowing the post title to be a paragraph, but I don't remember this being implemented? It was only implemented for the site title.

Because of the discussions in this issue https://github.com/WordPress/gutenberg/issues/30549
I have removed the partial support for p tag.